diff --git a/dashboard/index.html b/dashboard/index.html
index 3bcf80d..f2d9acc 100644
--- a/dashboard/index.html
+++ b/dashboard/index.html
@@ -3,10 +3,10 @@
 <head>
     <meta charset="UTF-8">
     <title>Best Damn Roomie Dashboard</title>
-    <link rel='stylesheet prefetch' href='http://fonts.googleapis.com/css?family=Open+Sans:400,300'>
-    <link rel='stylesheet prefetch' href='http://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css'>
+    <link rel='stylesheet prefetch' href='https://fonts.googleapis.com/css?family=Open+Sans:400,300'>
+    <link rel='stylesheet prefetch' href='https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css'>
     <link rel="stylesheet" href="css/style.css">
-    <script src='http://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js'></script>
+    <script src='https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js'></script>
 </head>
 
 <body>
@@ -42,7 +42,7 @@
         <div class="brick closed" style="background: transparent;min-width: 240px;min-height: 240px;">
             <hgroup>
                     <button type="button" id="steam" style="padding: 0;border: none;background: none;">
-                        <img src="/images/Steam_logo.png" style="width: 100%">
+                        <img src="images/Steam_logo.png" style="width: 100%">
                     </button>
             </hgroup>
         </div>
@@ -50,7 +50,7 @@
         <div class="brick closed" style="background: transparent;min-width: 240px;min-height: 240px;">
             <hgroup>
                     <button type="button" id="kodi" style="padding: 0;border: none;background: none;">
-                        <img src="/images/kodi_logo.png" style="width: 100%">
+                        <img src="images/kodi_logo.png" style="width: 100%">
                     </button>
             </hgroup>
         </div>
@@ -58,7 +58,7 @@
         <div class="brick closed" style="background: transparent;min-width: 240px;min-height: 240px;">
             <hgroup>
                     <button type="button" id="netflix" style="padding: 0;border: none;background: none;">
-                        <img src="/images/Netflix_logo.png" style="width: 100%">
+                        <img src="images/Netflix_logo.png" style="width: 100%">
                     </button>
             </hgroup>
         </div>
diff --git a/dashboard/js/index.js b/dashboard/js/index.js
index 0de6836..87ce2b9 100644
--- a/dashboard/js/index.js
+++ b/dashboard/js/index.js
@@ -6,7 +6,7 @@ $( "#steam" ).click(function() {
   var error = function(error) {
 	alert("An error happened");
   };
-  $.get( "http://127.0.0.1:5000/dashboard/switch/steam", success, error);
+  $.get( "/dashboard/switch/steam", success, error);
 });
 
 
diff --git a/mediadashboard.py b/mediadashboard.py
index a98b73b..b5cc76d 100644
--- a/mediadashboard.py
+++ b/mediadashboard.py
@@ -21,9 +21,11 @@ def startstuff(program):
 @app.route('/dashboard/switch/<process>')
 def setcurrentwindow(process):
     x = ['Wireshark', 'steam', 'kodi']
+
     if process in x:
         return setactivewindow(process)
     else:
         return 'I\'m sorry Dave, I\'m afraid I can\'t do that'
+
 if __name__ == '__main__':
-    app.run(debug =True)
+    app.run(debug =True, host='0.0.0.0')
diff --git a/requirements.txt b/requirements.txt
index c64d74e..9e79f40 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,4 +1,3 @@
 flask
 flask-cors
-python 2.7.3
-
+#python 2.7.3
diff --git a/setactivewindow.py b/setactivewindow.py
index 9a679f8..3df5bc1 100644
--- a/setactivewindow.py
+++ b/setactivewindow.py
@@ -5,7 +5,10 @@ def setactivewindow( process ):
     #activewindow = subprocess.check_output('xdotool getactivewindow', shell=True).strip()
     #activewindow = hex(int(activewindow))
 
-    active = subprocess.check_output('wmctrl -lp', shell=True)
+
+    active = subprocess.check_output(['wmctrl', '-lp'], env={'DISPLAY': ':0'})
+
+    print(active)
     activelines = active.splitlines()
     #activearray = []
 
@@ -14,7 +17,7 @@ def setactivewindow( process ):
             print 'process found'
             print line
             print 'windowactivate input=' + hex(int(line.split(None,1)[0], 16))
-            subprocess.check_output('xdotool windowactivate ' + hex(int(line.split(None,1)[0], 16)), shell=True).strip()
+            subprocess.check_output('xdotool windowactivate ' + hex(int(line.split(None,1)[0], 16)), shell=True, env={'DISPLAY': ':0'}).strip()
             return("success");
         else:
             return("process not started")
diff --git a/startprocess.py b/startprocess.py
index b83cd44..35d7144 100644
--- a/startprocess.py
+++ b/startprocess.py
@@ -3,9 +3,9 @@ import subprocess
 def startprocess( program ):
 
     try:
-        subprocess.check_call('pgrep ' + program + ' > /dev/null', shell=True )
+        subprocess.check_call('pgrep ' + program + ' > /dev/null', shell=True, env={'DISPLAY': ":0"} )
         print program + ' is running'
         return False
     except:
-        subprocess.Popen(program + ' &', shell=True)
+        subprocess.Popen(program + ' &', shell=True, env={'DISPLAY': ":0"})
         return True
